src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -508,7 +516,9 @@ void StorageKafka::threadFunc()
 
     // Wait for attached views
     if (!stream_cancelled)
-        task->scheduleAfter(RESCHEDULE_MS);
+    {
+        tasks[idx]->holder->scheduleAfter(RESCHEDULE_MS);
```

Comment:
Once separate pool will be introduced, will this be required? Hm, more jobs that the pool size is still possible, but AFAIR there are some issues if you are not polling kafka for some period

Worth thinking about it, maybe warn/reject creating more consumers then the pool size?